CONTRIBUTING.rst

```diff
@@ -105,19 +142,165 @@ This infrastructure is thoroughly described in this `wiki_section <https://githu
+Last but not least, even if a PR has been recently closed and/or merged, feedback would still be welcomed.
+
+Creating unit tests
+-----------------------
+
+Test files need to be located under WMCore/test/python/ and
+need to mirror the structure you use for the packages under
+WMCore/src/python where every directory and test file is
+augmented with a _t and the class should be augmented with 'Test'
+If you develop for an external package you mimic this structure
+in the external package directory structure.
+
+E.g.: if you have a source package::
+
+   src/python/WMCore/DataStructs/LumiList.py
+
+you will have a test in the package::
+
+   test/python/WMCore_t/DataStructs_t/LumiList_t.py
+
+and in this module you would define a class named ``LumiListTest``, which inherits from ``unittest.TestCase``.
+
+Your test class is supposed to implement the following methods:
+
+- ``setUp``: this method implements a pre-setup for every single unit test (e.g., the database schema definition).
+- ``tearDown``: this method implements a post-setup for every single unit test that gets executed, regardless of its exit code (e.g., cleaning up the database).
+- ``test*``: methods starting with the ``test`` word are automatically executed by the unittest framework (and Jenkins).
+
+Potential log files for tests should have a name: ``<testfile>.log``.
+So in case of the ``LumiList_t.py`` you would get a ``LumiList_t.log`` file.
 
 Coding Style and checks
 -----------------------
 
-We use pep8 and pylint (including pylint3) to sanitize our code. Please do the same before submitting a pull request.
-WMCore defines its own pylintrc `here <https://github.com/dmwm/WMCore/blob/master/standards/.pylintrc>`_, thus you should always pass this file as an argument when running pylint locally.
+We use pep8 and pylint to sanitize our code. Please do the same before submitting a pull request.
+WMCore defines its own pylintrc `here <https://github.com/dmwm/WMCore/blob/master/standards/.pylintrc>`_ standards.
+Thus, when evaluating your changes, please run pylint by passing this pylintrc file in the command line, your code should get scored 8 or above.
+Unless there is a very good reason, we discourage the use of pylint disable statements.
 
 Extra documentation
 -------------------
 
 In case you're having issues with git and working through a branch feature, you might want to have a look at this old'ish `wiki <https://github.com/dmwm/WMCore/wiki/Developing-against-WMCore>`_ in our WMCore wiki documentation.
 In addition to that, we've also compiled a long list of important git `commands <https://github.com/dmwm/WMCore/wiki/git-commands>`_. If none of those work for you, google and stackoverflow will be your best friend.
 
+
+Structural and in-depth documentation
+============
+
+WMCore structure
+-------------------
```

Comment:
I would say every WMCore/WMAgent path explained in this paragraph here is worth a label in bold.

Reply:
They have been marked as a code block, so I'm afraid I cannot make them bold.

Regarding `wmcore-new-config` and `DefaultConfig.py`, I didn't set them to bold intentionally. Reason is that I have never used that and it could be outdated, at the same time, I didn't want to remove it before making sure it was the case.